tests/basic/pthread_detach/README.md

```diff
@@ -0,0 +1,181 @@
+# Preliminaries
+## Thread identities
+A thread has 3 identities:
+- Linux process context - `task_struct`
+- LKL arch specific process context - `thread_info`
+- LKL host thread context - `lthread`
+
+For most of a thread's lifetime these 3 identities should have a 1:1 mapping between them.
```

Comment:
It's not quite bijective.  Not every lthread corresponds to an LKL thread.  In conventional LKL usage, the Linux process contexts are allocated lazily on first system call.  In cloned threads, the Linux process context is allocated eagerly.

Reply:
Added comment to doc.